server/src/main/java/org/opensearch/cluster/routing/allocation/AllocateUnassignedDecision.java

```diff
@@ -105,7 +105,7 @@ public class AllocateUnassignedDecision extends AbstractAllocationDecision {
     private final long remainingDelayInMillis;
     private final long configuredDelayInMillis;
 
-    private AllocateUnassignedDecision(
+    public AllocateUnassignedDecision(
```

Comment:
Can we retain it to be private and add static methods for constructing Decision ?

Reply:
Yes, made changes and pushed